src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,43 +1,79 @@
-import React from 'react';
+import React, { useContext, useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import classNames from 'classnames';
+import { TodoContext } from '../../contexts/TodoContext';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
 
 export const TodoModal: React.FC = () => {
+  const { currentTodo, setCurrentTodo } = useContext(TodoContext);
+  const [currentUser, setCurrentUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    if (currentTodo) {
+      setIsLoading(true);
+      getUser(currentTodo.userId)
+        .then(setCurrentUser)
+        .finally(() => {
+          setIsLoading(false);
+        });
+    }
```

Comment:
Try out your site; after you've opened a todo once, every time you try to open a new one, it pops up and then the loading starts; this could be avoided by resetting `isLoading` as soon as the modal closes
```js
if (currentTodo) {...}
else {
  setIsLoading(true);
}
```

This could also be achieved by resetting current user upon changing current todo and updating the loading condition to `{isLoading || !currentUser) ? <Loader /> : ...}`

Reply:
Done!